source/views/streaming/streams/list.tsx

```diff
@@ -23,99 +24,137 @@ const styles = StyleSheet.create({
 	},
 })
 
-export const StreamListView = (): JSX.Element => {
-	let [error, setError] = React.useState<Error | null>(null)
-	let [loading, setLoading] = React.useState(true)
-	let [refreshing, setRefreshing] = React.useState(false)
-	let [streams, setStreams] = React.useState<
-		Array<{title: string; data: StreamType[]}>
-	>([])
+let groupStreams = (entries: StreamType[]) => {
+	let grouped = groupBy(entries, (j) => j.$groupBy)
+	return toPairs(grouped).map(([title, data]) => ({title, data}))
+}
 
-	React.useEffect(() => {
-		try {
-			getStreams().then(() => {
-				setLoading(false)
-			})
-		} catch (error) {
-			if (error instanceof Error) {
-				setError(error)
-			} else {
-				setError(new Error('unknown error - not an Error'))
-			}
-			return
+let filterStreams = (entries: StreamType[], filters: ListType[]) => {
+	return entries.filter((stream) => {
+		let enabledCategories = filters.flatMap((f: ListType) =>
+			f.spec.selected.flatMap((s) => s.title),
+		)
+
+		if (enabledCategories.length === 0) {
+			return entries
 		}
-	}, [])
 
-	let refresh = async (): Promise<void> => {
-		setRefreshing(true)
-		await getStreams(true)
-		setRefreshing(false)
-	}
+		return enabledCategories.includes(stream.category)
+	})
+}
 
-	let getStreams = async (
-		reload?: boolean,
-		date: Moment = moment.tz(timezone()),
-	) => {
-		let dateFrom = date.format('YYYY-MM-DD')
-		let dateTo = date.clone().add(2, 'month').format('YYYY-MM-DD')
-
-		let data = await fetch(API('/streams/upcoming'), {
-			searchParams: {
-				sort: 'ascending',
-				dateFrom,
-				dateTo,
-			},
-			delay: reload ? 500 : 0,
-		}).json<Array<StreamType>>()
-
-		data = data
-			.filter((stream) => stream.category !== 'athletics')
-			.map((stream) => {
-				let date: Moment = moment(stream.starttime)
-				let dateGroup = date.format('dddd, MMMM Do')
-
-				let group = stream.status.toLowerCase() !== 'live' ? dateGroup : 'Live'
-
-				return {
-					...stream,
-					// force title-case on the stream types, to prevent not-actually-duplicate headings
-					category: titleCase(stream.category),
-					date: date,
-					$groupBy: group,
-				}
-			})
-
-		let grouped = groupBy(data, (j) => j.$groupBy)
-		let mapped = toPairs(grouped).map(([title, data]) => ({title, data}))
-
-		setStreams(mapped)
-	}
+let useStreams = (date: Moment = moment.tz(timezone())) => {
+	let dateFrom = date.format('YYYY-MM-DD')
+	let dateTo = date.clone().add(2, 'month').format('YYYY-MM-DD')
+
+	return useFetch<StreamType[]>(
+		API('/streams/upcoming', {
+			sort: 'ascending',
+			dateFrom,
+			dateTo,
+		}),
+		{
+			headers: {accept: 'application/json'},
+		},
+	)
+}
 
-	let keyExtractor = (item: StreamType) => item.eid
+export const StreamListView = (): JSX.Element => {
+	let {data = [], error, reload, isPending, isInitial, isLoading} = useStreams()
```

Comment:
Can we give this an interface? These fields seem magical without a type.

Reply:
It does look magical up-front but I guarantee that the types are all pretty clear since this now uses [react-async's `useFetch`](https://docs.react-async.com/api/interfaces?q=usefetc#usefetch-hook).

The type info on line 63 for `useStreams` (or up above on line 46) shows the following:

```ts
let useStreams: (date?: Moment) => AsyncState<StreamType[], FetchRun<StreamType[]>>
```

and hovering over each of these types reflects the [type information that `useFetch`'s state](https://docs.react-async.com/api/state) provides us

```tsx
let data: StreamType[]
let error: Error | undefined
let reload: () => void
let isPending: boolean
let isInitial: false
let isLoading: boolean
```